File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonModel/src/main/java/triathlon/model/Referee.java
package triathlon.model;

import jakarta.persistence.Column;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.io.Serializable;

@jakarta.persistence.Entity
public class Referee implements Entity<Long>, Serializable {
    @Id
    @GeneratedValue(generator = "increment")
    private Long id;
    @Column(name = "nameReferee")
    private String name;
    @Column
    private String username;
    @Column
    private String password;

    public Referee() {
    }

    public Referee(String name, String username, String password) {
        this.name = name;
        this.username = username;
        this.password = password;
    }

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long aLong) {
        this.id = aLong;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public String toString() {
        return "Referee{" +
                "name='" + name + '\'' +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                ", id=" + id +
                '}';
    }
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/model/Referee.cs
using System;
using Newtonsoft.Json;

namespace model;

[Serializable]
public class Referee(string name, string username, string password) : Entity<long>
{
    [JsonProperty("name")]
    public string Name { get; set; } = name;
    [JsonProperty("username")]
    public string Username { get; set; } = username;
    [JsonProperty("password")]
    public string Password { get; set; } = password;

    public override string ToString()
    {
        return base.ToString() + $", Name: {Name}, Username: {Username}, Password: {Password}";
    }
}